src/main/java/org/opensearch/sdk/handlers/ExtensionActionResponseHandler.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.handlers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.extensions.ExtensionsManager;
+import org.opensearch.extensions.action.ExtensionActionResponse;
+import org.opensearch.sdk.SDKTransportService;
+import org.opensearch.common.Nullable;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class handles the response from OpenSearch to a {@link SDKTransportService#sendProxyActionRequest()} call.
+ */
+public class ExtensionActionResponseHandler implements TransportResponseHandler<ExtensionActionResponse> {
+
+    private static final Logger logger = LogManager.getLogger(ExtensionActionResponseHandler.class);
+    private final CompletableFuture<ExtensionActionResponse> inProgressFuture;
+    private byte[] responseBytes = null;
+
+    /**
+    * Instantiates a new ExtensionActionResponseHandler
+    */
+    public ExtensionActionResponseHandler() {
+        this.inProgressFuture = new CompletableFuture<>();
+    }
+
+    @Override
+    public void handleResponse(ExtensionActionResponse response) {
+        logger.info("Received response bytes: " + response.getResponseBytes().length + " bytes");
+
+        logger.debug("Received response bytes: " + response.getResponseBytesAsString());
```

Comment:
Any specific reason to have info/debug together?

Reply:
The bottom one was a debug string while things weren't working.  It's gone already.